### ISSUE TITLE
Problem downloading hosts file from the "hblock" project.

### ISSUE THREAD

**Issue by tuqueque** [IngoZenz/personaldnsfilter#132]
Hey, first of all... THANK YOU for this wonderful app!

I've tried this app before with mixed results, but a couple of days ago I tried it again and so far has been GREAT!... Besides, is the only DNS filter app compatible with old versions of Android. This is the only app I was able to install in an old Android 4.2.2 tablet and is doing exactly what I wanted!

Now, the issue... In that old tablet, when I try to download the hosts file from https://hblock.molinero.dev/hosts (this is the unified hosts file from the "hblock" project)... I get this error:

```
DNS Filter: Reloading hosts filter...
Acquired WIFI lock and partial wake lock!
ERROR loading filter: https://hblock.molinero.dev/hosts
Released WIFI lock and partial wake lock!
Cannot update hosts file!
java.net.ConnectException: failed to connect to hblock.molinero.dev/hosts/2a01:4f8:c17:f260::1 (port 443) after 120000ms:
connect failed ENETUNREACH (Network is unreachable)
Retry at Tue Dec 29 12:13:42 GMT-04:30 2020
```

This only happens in that old tablet... when I use PersonalDNSfilter on my newer Android 9.0 phone, the hosts file downloads without any issue.

BTW, I also noticed that in the old tablet, I can't access the whitelist feature... I guess that's a limitation of the old Android version in the tablet.

Greets and happy new year!

**Comment by ryder203** [comment]
Hi. I can not reproduce this error. I added https://hblock.molinero.dev/hosts to my other filterlist sources and set it active, restarted/reloaded filters, all works, all gets loaded. As you write you got problems on a rather old device with Android 4.2.2 - have [you seen this](https://github.com/IngoZenz/personaldnsfilter/issues/129)?

**Comment by tuqueque** [comment]
It isn't Android 4.2.2 but 4.4.2, sorry...

I worked around the problem by going through the list in: https://github.com/hectorm/hblock/blob/master/SOURCES.md and adding each individual source hosts URL from the secondary mirror (the ones that begin with "https://raw.githubusercontent.com...")... Because the first URLs also gave me problems to download.

So I guess it is some sort of problem with the way it connects to those locations. I know nothing about those technical stuff, but that SSL problem from the other issue report you referenced might also be the problem here... Either way, I made it work with the workaround I described above.

BTW, would it be possible to add some sort of "blacklist" in PersonalDNSfilter to block specific apps from accessing the Internet altogether through either WiFi or the phone's data plan?... That's the only feature I'd like to have to completely replace NetGuard ( https://github.com/M66B/NetGuard )

Thank you for the quick reply on my issue!

**Comment by IngoZenz** [comment]
Seem the issue is not related to SSL but some other issue? Wondering why it is trying to connect via IPV6 IP?
Can you open the URL vis browser on this device?

**Comment by tuqueque** [comment]
> Seem the issue is not related to SSL but some other issue? Wondering why it is trying to connect via IPV6 IP?
> Can you open the URL vis browser on this device?

Yes, I can open https://hblock.molinero.dev/hosts in the tablet's browser without any issue.

I thought the same about PersonalDNSfilter attaching that IPv6 address at the end of the URL, but since I know nothing about these things, I thought it was "normal".

And of course, if I attach the IPv6 address to the URL (hblock.molinero.dev/hosts/2a01:4f8:c17:f260::1) the browser just hangs there forever trying to connect.

**Comment by IngoZenz** [comment]
it is Android here adding the IPv6 ip to the URL when generating the connect error message... it is a bit confusing. But the main issue seems to be that the V6 IP is used ...

**Comment by tuqueque** [comment]
> it is Android here adding the IPv6 ip to the URL when generating the connect error message... it is a bit confusing. But the main issue seems to be that the V6 IP is used ...

Any suggestions?... I've looked in the tablet's system settings everywhere and I don't see anything related to "forcing" IPv6 connections or stuff like that.

**Comment by IngoZenz** [comment]
Is the issue still present?

**Comment by tuqueque** [comment]
I can't say. It was on a friend's old Android tablet. I'll see if I can check again over the weekend.

**Comment by tuqueque** [comment]
I just tested this again (having updated personalDNSfilter to the latest version in F-droid first, of course) and the problem remains... It throws the exact same error.

As I mentioned in one of the comments before, at least adding each individual list in https://github.com/hectorm/hblock/blob/master/SOURCES.md works. So is not all lost.

Thanks again for keep improving this fantastic app anyway! This is by far my favorite DNS filter and ad/tracker blocker!

**Comment by IngoZenz** [comment]
could you add the following lines to additional hosts, restart pDNSf and try again?

```
>hblock.molinero.dev 116.203.252.132
>hblock.molinero.dev  ::
```

In case this is not working, could you provide the trafficlog from teh personaldnsfilter log folder?

**Comment by tuqueque** [comment]
Still not working... I'm attaching both, the "additional hosts" file and the traffic log so you can check I did it correctly.
file:///mnt/40C5294F094B8BCA/downloads/videos/1/PersonalDNSFilter/additionalHosts.txt
file:///mnt/40C5294F094B8BCA/downloads/videos/1/PersonalDNSFilter/log/trafficlog/trafficlog_0.log
